src/main.ts

```diff
@@ -145,6 +145,48 @@ export async function runAction(): Promise<any> {
         )
       )
     }
+
+    core.debug("Processing pipeline report...")
+    const failedMessage =
+      "Some pipeline actions have failed. Please check the pipeline report for details."
+    if (result && !result["passed"]) {
+      core.info(ansi.red(failedMessage))
+    }
+
+    if (
+      !Object.values(constants.EndStates).includes(executionGraph["status"])
+    ) {
+      core.setFailed(`Execution graph ${executionGraphId} has timed out.`)
+    } else {
+      if (executionGraph["status"] !== constants.EndStates.SUCCEEDED) {
+        displayErrorExecutionGraph(executionGraph)
+        core.setFailed(
+          `Execution graph ${executionGraphId} has ${executionGraph[
+            "status"
+          ].toLowerCase()}.`
+        )
+      } else {
+        core.debug(
+          `Execution graph ${executionGraphId} has completed successfully.`
+        )
+      }
+    }
+
+    const failedExecution = "true"
```

Comment:
This does not seem right. I might be missing something but it looks like this will mark all executions as failed. You should be using a boolean as well.